### PR TITLE
Add `defqueries`, upgrade Ecto to 3, and compile in documentation

### PR DESCRIPTION
**Pull request by zph**
Hello :D. Thanks for writing Elixir's implementation of yesql.

I implemented https://github.com/lpil/yesql/issues/3 as `Yesql.defqueries` in order to leave backwards compatibility alone but add this functionality. The reason for wanting to do so is that there's a good benefit to grouping similar queries into a single file vs one file per query. It removes boilerplate and questions of "did I reference that sql file already" each time you add a query.

This PR does the following:
- Adds `defqueries` that allows for alternate (multi-per-file syntax matching clojure's yesql)
- Upgrades to Ecto 3
- Compiles in documentation
- Uses `@external_resource` to ensure recompilation when SQL changes in `iex`

In a future PR, I'd like to switch to using a Tokenizer for splitting defqueries into component blocks, but this works and I'd rather contribute this now than wait 👍 .

### CODE REVIEW

File: mix.lock [hunk 1]
```diff
@@ -12,5 +13,6 @@
...
   "poolboy": {:hex, :poolboy, "1.5.1", "6b46163901cfd0a1b43d692657ed9d7e599853b3b21b95ae5ae0a777cf9b6ca8", [:rebar], [], "hexpm"},
-  "postgrex": {:hex, :postgrex, "0.13.5", "3d931aba29363e1443da167a4b12f06dcd171103c424de15e5f3fc2ba3e6d9c5", [:mix], [{:connection, "~> 1.0", [hex: :connection, repo: "hexpm", optional: false]}, {:db_connection, "~> 1.1", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 1.0", [hex: :decimal, repo: "hexpm", optional: false]}], "hexpm"},
+  "postgrex": {:hex, :postgrex, "0.15.0", "dd5349161019caeea93efa42f9b22f9d79995c3a86bdffb796427b4c9863b0f0", [:mix], [{:connection, "~> 1.0", [hex: :connection, repo: "hexpm", optional: false]}, {:db_connection, "~> 2.1", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 1.5", [hex: :decimal, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
+  "telemetry": {:hex, :telemetry, "0.4.0", "8339bee3fa8b91cb84d14c2935f8ecf399ccd87301ad6da6b71c09553834b2ab", [:rebar3], [], "hexpm"},
```
Comment (zph) [inline mix.lock:17]: Included w/ Ecto 3

File: test/yesql_test.exs [hunk 2]
```diff
@@ -128,6 +241,67 @@ defmodule YesqlTest do
     end
   end
 
+  describe "defqueries/2" do
```
Comment (zph) [inline test/yesql_test.exs:244]: These are duplicated in full from `defquery` to make sure I was covering the same circumstances.

**Comment by lpil** [comment]
Hi @zph , there's a lot of changes here, none of which have been discussed previously. Could you submit one PR per feature/change please and we'll work out how to go forward. Thanks :)

**Comment by lpil** [comment]
Closing due to inactivity

**Comment by zph** [comment]
@lpil Woops, I lost track of your response here. Thank you for the response and :).

Do you have an interest in all/any of these features if I circle back to package them up one at a time in PRs?

* Adds defqueries that allows for alternate (multi-per-file syntax matching clojure's yesql)
* Compiles in documentation
* Uses @external_resource to ensure recompilation when SQL changes in iex

**Pull request closed without merging**